### PR TITLE
Bugfix/slate value state when changing domain

### LINKED ISSUE #29: Changing Slider domain and value props at the same time creates error in setValues
### Problem or feature description:
Changing the domain and value props of the slider to a domain that is smaller can cause an "invalid value" warning from the slider.

This is reproducible with the following codesandbox: https://codesandbox.io/s/5woq82rz6k

The root cause appears to be here: https://github.com/sghall/react-compound-slider/blob/master/src/Slider/Slider.js#L82

In this control flow, the conditional on line 68 evaluates to true which causes the range to be updated, and onChange and onUpdate to fire, but the code continues to call `setValues` on line 84 using a stale set of values. This results in the slider ignoring the value prop that is actually being set, and it uses the old value set which is often out of range. The value is then reset to an incorrect value.

### Steps to reproduce (for problems):

1. Mount a slider with a domain and value set.
2. Change the domain and values at the same time to a domain smaller than the original props.
3. Observe that an error is logged to the console stating the range was reset.

### Versions (for problems):

React-Compound-Slider: 0.15.1

React: 16.

Browser: Chrome 

Operating System: Linux

### PR DESCRIPTION
**Pull request by CameronAckermanSEL**
Fixes #29

### CODE REVIEW

**Comment by CameronAckermanSEL** [comment]
Test coverage doesn't cover the control flow where values from props is undefined and reMapValues falls back to using state. My initial reaction is that maybe we don't need to fall back onto state at all. But I'm not sure.

File: src/Slider/Slider.js [hunk 1]
```diff
@@ -98,9 +99,9 @@ class Slider extends PureComponent {
     }
   }
 
-  reMapValues(reversed) {
-    const { values } = this.state
-    return this.setValues(values.map(d => d.val), reversed)
+  reMapValues(reversed, values) {
+    // if values was not passed, fall back to using state
+    return this.setValues(values || this.state.values.map(d => d.val), reversed)
```
Comment (CameronAckermanSEL) [inline src/Slider/Slider.js]: Test coverage doesn't cover the control flow where values from props is undefined and reMapValues falls back to using state. My initial reaction is that maybe we don't need to fall back onto state at all. But I'm not sure.

Comment (CameronAckermanSEL) [reply src/Slider/Slider.js]: A hint for how to cover this in tests would be welcome :), passing undef values props seems to not be the right way to accomplish this

Comment (sghall) [reply src/Slider/Slider.js]: This looks good.  I see what you mean about the tests.  I'll fix that up.  I was going to fix some things on the tests anyways.

**Pull request merged**